### PR TITLE
Adds the engiborg plasma tank manipulator module

### PR DESCRIPTION
**Pull request by PatchouliKnowledge**
It has come to my attention that the engineering borgs can't set up the singularity due to not having arms to manipulate the plasma tanks needed. Little drones chuckle and set up the singulo while engiborgs shed tears of humiliation and wire solars. But not anymore!
- Adds a new engiborg module - plasma tank manipulator. It picks up plasma tanks from the world, and inserts them into portable atmos machinery/canisters/radiation collectors.
- Also gives borgs the ability to dispense tanks from tank dispensers (such as the one in engineering storage).

### CODE REVIEW

**Comment by MrPerson** [comment]
Wouldn't it make more sense to just change the plasma collectors? Hell you could remove the whole need to replace the tanks outright and nobody would care

**Comment by PatchouliKnowledge** [comment]
Changing the collectors to not require plasma would take away a very important gameplay element, that is, novice engineers flooding the storage with plasma.

File: code/modules/power/singularity/collector.dm [hunk 1]
```diff
@@ -116,21 +135,23 @@ var/global/list/rad_collectors = list()
...
 	Z.loc = get_turf(src)
 	Z.layer = initial(Z.layer)
-	src.P = null
+	src.ptank = null
+	Z = null
 	if(active)
 		toggle_power()
 	else
 		update_icons()
+	return
```
Comment (phil235) [inline code/modules/power/singularity/collector.dm]: why can't this proc use ptank directly instead of a Z var?

**Comment by KorPhaeron** [comment]
Robots taking our jobs

File: code/game/objects/items/weapons/tankmanipulator.dm [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+			storedtanks.Remove(T)
+			T.loc = D
+			D.plasmatanks++
+			tanks--
+		playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
+		D.update_icon()
+		D.updateUsrDialog()
+		update_icon()
+		user << "<span class='notice'>You unload some tanks into the dispenser.</span>"
+		return
```
Comment (phil235) [inline code/game/objects/items/weapons/tankmanipulator.dm]: why make three different afterattack proc?

File: code/game/objects/items/weapons/tankmanipulator.dm [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+			I.loc = src
+			tanks++
+			update_icon()
+			return
+		else
+			user << "<span class='warning'>The manipulator is full!</span>"
+			return
+	//Picks up tanks from the world
+
+/obj/item/weapon/tankmanip/afterattack(obj/machinery/M, mob/user, proximity)
```
Comment (GunHog) [inline code/game/objects/items/weapons/tankmanipulator.dm]: Is having three afterattack() procs for the same object valid? Does that work? It is not normally combined into one and just having each type check handle its own unique code? Also, there are some common lines in the code here you could combine.

**Comment by phil235** [comment]
It's a pretty snowflake module tbh, if at least it could grab all tanks I'd be in favor of it. But right now this is a single use module. There's plenty of things that engiborgs can't do, if we start adding a module for each special case we're going to end up with too many modules.

**Comment by PatchouliKnowledge** [comment]
@phil235 I have considered having the module pick up any type of tank; however, that is completely useless for engiborgs. They can't use internals tanks or anaesthetic tanks for anything whatsoever, only plasma tanks have any value since they can be used for the singularity engine (or the supermatter engine which uses collectors too), so there's no point in having that option.

**Comment by Miauw62** [comment]
why is this neccesary.
get somebody to help you.

**Comment by SconesC** [comment]
I don't think cyborgs should be able to do everything a human can

Setting up the engine is among these, mainly because it requires hands

Just go do solars or something cyborg

**Comment by PatchouliKnowledge** [comment]
The point of this is that the Engineering department isn't always staffed (yes, this happens even on Sybil). If it's not staffed, you go and do solars OR try and ask someone with access, i.e. HoP or Captain, good luck with that. If it is staffed, you go and do solars OR ask someone, though chances are they did it already.

Solars are no longer an option for stable power though - R&D builds so much shit that the power usage eventually goes above what solars can handle.

Furthermore, I should point out again that a drone can set up the engine, whereas a borg can't, and while this is unintentional, it's just silly. As for borgs doing what humans can, borgs are designed to be more efficient at their 'department' than a single human - this is only filling a small design gap.

**Comment by Jalleo** [comment]
They are more designed to not be able to work as anything outside of their department really.

**Comment by optimumtact** [comment]
Borgs are not designed to be an always better choice than humans, so I don't think we should support this.

**Comment by SconesC** [comment]
Restating what I said earlier for emphasis
>Borgs are not designed to be an always better choice than humans, so I don't think we should support this.

Also solars are 100% reliable what are you talking about

**Comment by Shadowlight213** [comment]
Yeah. Wiring the solars to the grid works out just fine. It may have trouble if the smes are dry and it has to charge the APCs, but otherwise i've never seen  the grid have issues.

**Comment by Midaychi** [comment]
How about, instead of a module, you just make tsnks clickdraggable into collectors?

**Comment by Fox-McCloud** [comment]
Why not just add a mousedrop to allow plasma tanks to be loaded in that way?

Also, removing plasma filling for collectors is a bad call, IMO; it allows engineers to experiment with lower and much higher power levels, depending on the state of the gas.

**Comment by Aranclanos** [comment]
well I like the idea but I'm not a fan of adding another object to the already clustered engineering borg tools list, can't there be another way? Plus I don't see that object being used for anything besides setting up the singularity.

**Comment by RemieRichards** [comment]
As has been suggested twice, just give collectors a mousedrop

**Comment by Aranclanos** [comment]
mousedrop is not really a common option and it won't be really known

**Comment by RemieRichards** [comment]
It's used for a few things, and the more we use it, the more known it will become.

**Comment by Aranclanos** [comment]
there's nothing that says 'this object accepts mousedrop'

**Comment by RemieRichards** [comment]
there's nothing that says "this object accepts clicks" "this objects accepts being clicked by a multitool" "this object accepts being clicked by a plasma tank" "this object can be emagged", etc.

**Comment by PatchouliKnowledge** [comment]
Ohkayy, so since this shaped up to be something completely different and now has merge conflicts too, I'll just close and rewrite this from scratch with suggestions taken into account. Later.

**Pull request closed without merging**